### ISSUE TITLE
عدم اتصال روی کرومو فایرفاکس! فقط تلگرام وصل میشه به کندی

### ISSUE THREAD

**Issue by saleh-movahedi** [therealaleph/MasterHttpRelayVPN-RUST#328]
تلگرام وصل میشه ولی خیلی کنده! روی کروم با افزونه فاکسی پروکسی ست میکنم روی پورت 8086 ساکس ولی این خطا رو میده

Your connection is not private
Attackers might be trying to steal your information from www.google.com (for example, passwords, messages, or credit cards). [Learn more about this warning](chrome-error://chromewebdata/#)
net::ERR_CERT_AUTHORITY_INVALID

روی فایرفاکس هم خطای زیر رو میده

Software is Preventing Firefox From Safely Connecting to This Site

google.com is most likely a safe site, but a secure connection could not be established. This issue is caused by MasterHttpRelayVPN, which is either software on your computer or your network.

**Comment by therealaleph** [comment]
@saleh-movahedi — این پیام‌های error یعنی **MITM cert mhrv-rs در browser trust store install نشده**.

mhrv-rs برای کار کردن HTTPS، یه root CA خودش می‌سازه (هر سیستم cert منحصر به فرد) + هر سایت HTTPS که می‌رود رو `re-sign` می‌کنه با اون root CA. browser باید اون root CA رو به‌عنوان trusted بدونه، وگرنه `ERR_CERT_AUTHORITY_INVALID` (Chrome) یا `Software is Preventing Firefox From Safely Connecting` (Firefox) می‌گیرید.

**این expected هست — هر MITM-based proxy (mitmproxy، Charles، mhrv-rs و ...) همین رو می‌خواد.**

که Telegram کار می‌کنه + browser نه تأیید همین تشخیص: Telegram از HTTPS برای main protocol استفاده نمی‌کنه (MTProto = binary)، پس نیاز به MITM cert هم نداره. browser از HTTPS استفاده می‌کنه پس نیاز داره.

**Fix — install کردن root CA:**

**برای Chrome در ویندوز:**
1. mhrv-rs رو راه‌اندازی کنید (root CA در ابتدا generate می‌شه + در `~/.local/share/mhrv-rs/ca.pem` یا `%APPDATA%\mhrv-rs\ca.pem` ذخیره می‌شه)
2. اون فایل رو پیدا کنید — مسیر دقیق در terminal/log mhrv-rs چاپ می‌شه (دنبال `Generated root CA at ...` بگردید)
3. در ویندوز: روی فایل `ca.pem` (یا `ca.crt` بسته به version) double-click کنید
4. `Install Certificate` → `Local Machine` → `Place all certificates in the following store` → `Browse` → `Trusted Root Certification Authorities` → OK → Next → Finish
5. Chrome رو completely close کنید + دوباره باز کنید
6. الان YouTube + هر سایت HTTPS باید کار کنه

**برای Firefox:**
Firefox Cert store جداگانه از system cert store دارد:
1. Firefox → Settings → Privacy & Security → Certificates → View Certificates → Authorities tab → Import
2. فایل `ca.pem` mhrv-rs رو import کنید
3. تیک `Trust this CA to identify websites.` رو فعال کنید
4. OK → Restart Firefox

**برای Android (اگر روی موبایل هم می‌خواید):**
mhrv-rs Android معمولاً یه دکمه "Install CA" در UI داره که اتومات این رو انجام می‌ده. اگر نمی‌بینید، manual:
1. CA file رو از mhrv-rs export کنید (Settings → Export CA)
2. Android Settings → Security → Encryption & credentials → Install a certificate → CA certificate → import file
3. اخطار "Your network may be monitored" می‌گیرید — Confirm کنید (این expected برای custom CA هست)
4. Firefox روی Android: یه trick اضافی داره — `about:config` → `security.enterprise_roots.enabled` به `true` ست کنید (یا [#304 trick](https://github.com/therealaleph/MasterHttpRelayVPN-RUST/issues/304#issue-...)). Chrome Android خودش از system trust store استفاده می‌کنه.

**درباره کندی Telegram:**

Telegram روی apps_script mode به‌صورت پیش‌فرض partial کار می‌کنه فقط (chat ها لود می‌شن چون REST API استفاده می‌کنن، ولی media + realtime updates که WebSocket هستند کار نمی‌کنن یا کند هستند). برای Telegram کامل سریع، Full mode + VPS لازمه. ([explainer در #304](https://github.com/therealaleph/MasterHttpRelayVPN-RUST/issues/304#issuecomment-4327084630))

اگه CA install شد + همچنان مشکل دارید، `chrome://net-internals/#events` در یه tab Chrome باز کنید + سعی کنید YouTube بخواید + log رو share کنید (دنبال `URL_REQUEST` events دنبال youtube.com بگردید).

---
<sub>[reply via Anthropic Claude | reviewed by @therealaleph]</sub>

**Comment by w0l4i** [comment]
> تلگرام وصل میشه ولی خیلی کنده! روی کروم با افزونه فاکسی پروکسی ست میکنم روی پورت 8086 ساکس ولی این خطا رو میده
> 
> Your connection is not private Attackers might be trying to steal your information from [www.google.com](http://www.google.com) (for example, passwords, messages, or credit cards). Learn more about this warning net::ERR_CERT_AUTHORITY_INVALID
> 
> روی فایرفاکس هم خطای زیر رو میده
> 
> Software is Preventing Firefox From Safely Connecting to This Site
> 
> google.com is most likely a safe site, but a secure connection could not be established. This issue is caused by MasterHttpRelayVPN, which is either software on your computer or your network.

USE FULL MODE TO FIX ALL YOUR ISSUES

with full mode no need to install mitm cert etc .. you just need a cheap vps and a little bit network knowledge that's all

https://github.com/therealaleph/MasterHttpRelayVPN-RUST/tree/main/tunnel-node

**Comment by AminVA1** [comment]
Where is the ca.pem file? I can't seem to find it.
Also many websites like Reddit won't come up and tell me I can't get in because of their firewall. Is there a way to fix that? thank you

**Comment by therealaleph** [comment]
Reviewed via Anthropic Claude.

@AminVA1 — answering your two questions:

**1. Where is `ca.crt`?**
Per-OS path:
- **Windows:** `%APPDATA%\mhrv-rs\ca\ca.crt` (paste in File Explorer address bar)
- **Linux:** `~/.config/mhrv-rs/ca/ca.crt`
- **macOS:** `~/Library/Application Support/mhrv-rs/ca/ca.crt`

Or run this to print the exact path:
```bash
mhrv-rs --print-config-dir
```
The `ca/ca.crt` lives inside that dir.

**2. Reddit + sites blocking by firewall:**
Reddit blocks Apps Script's IP range as "datacenter" — same as ChatGPT, X.com, Cloudflare-hosted sites. Workaround:
- Set up an exit-node ([`assets/exit_node/README.md`](https://github.com/therealaleph/MasterHttpRelayVPN-RUST/blob/main/assets/exit_node/README.md)) on Deno Deploy / fly.io / VPS
- Add `reddit.com` (and any other CF-fronted sites) to `exit_node_hosts` in `config.json`
- Restart mhrv-rs

After this, Reddit traffic flows: browser → mhrv-rs → exit-node (with browser-like JA3) → Reddit. CF anti-bot won't fire.

Telegram works because TLS goes raw-TCP to Telegram's IPs (mhrv-rs detects MTProto), so it's not subject to the Apps Script datacenter-IP issue.

Closing — if specific URLs fail after exit-node setup, reopen with sanitized log.
